test/gtest/cfg.cpp

```diff
@@ -290,3 +290,43 @@ TEST_F(CFGTest, FinitePowersetLatticeFunctioning) {
   element2.print(ss);
   EXPECT_EQ(ss.str(), "101101");
 }
+
+TEST_F(CFGTest, BlockIndexes) {
+  auto moduleText = R"wasm(
+    (module
+      (func $foo
+        (if
+          (i32.const 1)
+          (block
+            (drop
+              (i32.const 2)
+            )
+            (drop
+              (i32.const 3)
+            )
+          )
+        )
+      )
+    )
+  )wasm";
+
+  Module wasm;
+  parseWast(wasm, moduleText);
+
+  auto* func = wasm.getFunction("foo");
+  CFG cfg = CFG::fromFunction(func);
+  cfg.computeExpressionBlockIndexes();
+
+  // The body of the function is an if. An if is a control flow structure and so
+  // it has no basic block (it can contain multiple ones).
```

Comment:
For the analysis framework, we'll eventually want to do control-sensitive analyses and be able to quickly tell what conditions must be true given that we take a certain control flow edge. Knowing that we executed an `if` as opposed to something else will be important in that context. Also, `if` consumes a value from the stack and that will be important to model once we do analyses that track more than just values in locals. Will it cause problems if we update the CFG utility to include `if`s in basic blocks in the future?

Reply:
It wouldn't be a problem in anything I current have planned. Yes, I can see how we'd want to connect ifs to basic blocks in some manner (and the same for try, I think).